e2e/pos_test.go

```diff
@@ -0,0 +1,144 @@
+package e2e
+
+import (
+	"context"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/0xPolygon/minimal/consensus/ibft/proto"
+	"github.com/0xPolygon/minimal/crypto"
+	"github.com/0xPolygon/minimal/e2e/framework"
+	"github.com/0xPolygon/minimal/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func findValidatorByAddress(validators []*proto.Snapshot_Validator, addr string) *proto.Snapshot_Validator {
+	for _, v := range validators {
+		if v.Address == addr {
+			return v
+		}
+	}
+	return nil
+}
+
+func TestPoS_Stake(t *testing.T) {
+	signer := &crypto.FrontierSigner{}
+	stakerKey, stakerAddr := framework.GenerateKeyAndAddr(t)
+	stakingContractAddr := types.StringToAddress("1001")
```

Comment:
We should not be hardcoding these values anywhere, you can grab them as constants from the system runtime

Reply:
Fixed!